crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -991,13 +991,12 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                         Some(url) => PubGrubDistribution::from_url(name, url),
                         None => PubGrubDistribution::from_registry(name, version),
                     };
-                    let version_id = dist.version_id();
 
                     // Wait for the metadata to be available.
                     self.index
                         .distributions()
-                        .wait_blocking(&version_id)
-                        .ok_or(ResolveError::Unregistered)?;
+                        .wait_blocking(&dist.version_id())
+                        .ok_or(ResolveError::Unregistered(dist.version_id().to_string()))?;
```

Comment:
Should these now be in a closure to avoid allocation?

Reply:
Good idea, will change